packages/@aws-cdk/aws-appsync/lib/schema.ts

```diff
@@ -1,27 +1,48 @@
 import { readFileSync } from 'fs';
-import { Lazy } from '@aws-cdk/core';
-import { CfnGraphQLSchema } from './appsync.generated';
-import { GraphqlApi } from './graphqlapi';
-import { SchemaMode, shapeAddition } from './private';
-import { IIntermediateType } from './schema-base';
-import { Field, ResolvableField } from './schema-field';
-import { ObjectType } from './schema-intermediate';
+import { IGraphqlApi } from './graphqlapi-base';
 
 /**
- * The options for configuring a schema
- *
- * If no options are specified, then the schema will
- * be generated code-first.
+  * Configuration for bound graphql schema
+  *
+  * Returned from ISchema.bind allowing late binding of schemas to graphqlapi-base
+  */
+export interface ISchemaConfig {
+  /**
+    * The ID of the api the schema is bound to
+    */
+  apiId: string;
+
+  /**
+    * The schema definition string
+    */
+  definition: string;
+}
+
+/**
+  * Interface for implementing your own schema
+  *
+  * Useful for providing schema's from sources other than assets
+  */
+export interface ISchema {
+  /**
+    * Binds a schema string to a GraphQlApi
+    *
+    * @returns ISchemaConfig with apiId and schema definition string
+    * @param api the api to bind the schema to
+    */
+  bind(api: IGraphqlApi): ISchemaConfig;
```

Comment:
```suggestion
  bind(api: IGraphqlApi, options?: SchemaBindOptions): ISchemaConfig;
```

So that we can modify this in the future without breaking changes?

Reply:
ahh yes, good call.